docs/RationaleSeq.md

```diff
@@ -106,3 +106,48 @@ information.
 - Initial value: this register is uninitialized. Using an uninitialized value
 results in undefined behavior. We will add an `initialValue` attribute if
 this proves insufficient.
+
+## The Register Operation
+
+`seq.reg` abstractly models a generic memory element that can be implemented
+as various physical memory elements. This attempts to capture most behaviours
+expressable by a `reg` in SystemVerilog.
+
+This is an abstract operation that should be further lowered down to either
+vendor-specific primtives or to the SV dialect.
+
+The register operation contains the following operands:
+
+- clock and its edge sensitivity: posedge, negedge, edge-triggered, or
+  unclocked. An unclocked register can be used to express a latch.
+- enable signals
+- input value
+- synchronous & asynchronus reset and reset values
+
```

Comment:
initial value?  Only applicable on some tagets, maybe, but those do care.  If not modeled by the op, how do you represent it?